src/ballet/shred/fd_shred.c

```diff
@@ -1,15 +1,125 @@
 #include "fd_shred.h"
 
 fd_shred_t const *
-fd_shred_parse( uchar const * buf ) {
+fd_shred_parse( uchar const * const buf,
+                ulong         const sz ) {
+  /* Reinterpret size as long: `sz` cannot realistically underflow.
+     `sz1` is the size of the entire shred. */
+  long const sz1 = (long)sz;
+
+  /* Initial bounds check */
+  if( FD_UNLIKELY( sz1<0x58L ) ) return NULL;
+
   fd_shred_t const * shred = (fd_shred_t *)buf;
-  /* Validate shred type */
-  uchar shred_type = fd_shred_type( shred->variant );
-  if( FD_UNLIKELY( shred_type!=FD_SHRED_TYPE_MERKLE_DATA &&
-                   shred_type!=FD_SHRED_TYPE_MERKLE_CODE &&
+
+  /* Validate shred type.
+     Safe to access because `variant` ends at 0x41, which is <= 0x58 */
```

Comment:
Any way to add a `static_assert( offsetof(shred, variant), 0x41)` here? Just a bit paranoid with parsing code.